### PR TITLE
FOGL-1027 - unit tests for services/core/scheduler/entities.py

### PR DESCRIPTION
**Pull request by amarendra-dianomic**
Test Result:
```
FogLAMP (FOGL-1027 *%) $ pytest -s -vv tests/unit/python/foglamp/services/core/scheduler/test_scheduler_entities.py 
======================================================= test session starts ========================================================
platform linux -- Python 3.5.2, pytest-3.4.0, py-1.5.2, pluggy-0.6.0 -- /usr/bin/python3.5
cachedir: .pytest_cache
rootdir: /home/asinha/Development/FogLAMP, inifile:
plugins: mock-1.6.0, cov-2.5.1, asyncio-0.8.0, allure-adaptor-1.7.10, aiohttp-0.3.0
collected 9 items                                                                                                                  

tests/unit/python/foglamp/services/core/scheduler/test_scheduler_entities.py::TestSchedulerEntities::test_ScheduledProcess PASSED
tests/unit/python/foglamp/services/core/scheduler/test_scheduler_entities.py::TestSchedulerEntities::test_Schedule PASSED
tests/unit/python/foglamp/services/core/scheduler/test_scheduler_entities.py::TestSchedulerEntities::test_schedule_todict PASSED
tests/unit/python/foglamp/services/core/scheduler/test_scheduler_entities.py::TestSchedulerEntities::test_StartUpSchedule PASSED
tests/unit/python/foglamp/services/core/scheduler/test_scheduler_entities.py::TestSchedulerEntities::test_TimedSchedule PASSED
tests/unit/python/foglamp/services/core/scheduler/test_scheduler_entities.py::TestSchedulerEntities::test_timedschedule_todict PASSED
tests/unit/python/foglamp/services/core/scheduler/test_scheduler_entities.py::TestSchedulerEntities::test_IntervalSchedule PASSED
tests/unit/python/foglamp/services/core/scheduler/test_scheduler_entities.py::TestSchedulerEntities::test_ManualSchedule PASSED
tests/unit/python/foglamp/services/core/scheduler/test_scheduler_entities.py::TestSchedulerEntities::test_Task PASSED

===================================================== 9 passed in 0.03 seconds =====================================================
FogLAMP (FOGL-1027 *%) ```
Code coverage:

```

Coverage for python/foglamp/services/core/scheduler/entities.py : 100% Show keyboard shortcuts
70 statements   70 run 0 missing 0 excluded

```

### CODE REVIEW

File: tests/unit/python/foglamp/services/core/scheduler/test_scheduler_entities.py [hunk 1]
```diff
@@ -0,0 +1,104 @@
+# -*- coding: utf-8 -*-
+
+# FOGLAMP_BEGIN
+# See: http://foglamp.readthedocs.io/
+# FOGLAMP_END
+
+""" Test foglamp/services/core/scheduler/entities.py """
+
+import uuid
```
Comment (Singhal-Vaibhav) [inline tests/unit/python/foglamp/services/core/scheduler/test_scheduler_entities.py]: unused import

File: tests/unit/python/foglamp/services/core/scheduler/test_scheduler_entities.py [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+        scheduled_process = ScheduledProcess()
+        assert scheduled_process.name is None
+        assert scheduled_process.script is None
+
+    def test_Schedule(self):
+        assert isinstance(Schedule.Type(2), IntEnum)
+
+        schedule = Schedule(Schedule.Type.STARTUP)
+        assert schedule.schedule_id is None
+        assert schedule.name  is None
```
Comment (Singhal-Vaibhav) [inline tests/unit/python/foglamp/services/core/scheduler/test_scheduler_entities.py]: multiple spaces before is

File: tests/unit/python/foglamp/services/core/scheduler/test_scheduler_entities.py [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+class TestSchedulerEntities:
+    def test_ScheduledProcess(self):
+        scheduled_process = ScheduledProcess()
+        assert scheduled_process.name is None
+        assert scheduled_process.script is None
+
+    def test_Schedule(self):
+        assert isinstance(Schedule.Type(2), IntEnum)
+
+        schedule = Schedule(Schedule.Type.STARTUP)
```
Comment (Singhal-Vaibhav) [inline tests/unit/python/foglamp/services/core/scheduler/test_scheduler_entities.py]: unused variable schedule

File: tests/unit/python/foglamp/services/core/scheduler/test_scheduler_entities.py [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+        startup_schedule = StartUpSchedule()
+        assert startup_schedule.schedule_id is None
+        assert startup_schedule.name  is None
+        assert startup_schedule.exclusive  is True
+        assert startup_schedule.enabled is False
+        assert startup_schedule.repeat is None
+        assert startup_schedule.process_name is None
+        assert startup_schedule.schedule_type == 1
+        with pytest.raises(AttributeError) as excinfo:
```
Comment (Singhal-Vaibhav) [inline tests/unit/python/foglamp/services/core/scheduler/test_scheduler_entities.py]: unused variable excinfo

File: tests/unit/python/foglamp/services/core/scheduler/test_scheduler_entities.py [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+            assert startup_schedule.day is None
+            assert startup_schedule.time is None
```
Comment (Singhal-Vaibhav) [inline tests/unit/python/foglamp/services/core/scheduler/test_scheduler_entities.py]: These assertions are never called. If you replace `is None` with `is not None` test passes then too?

File: tests/unit/python/foglamp/services/core/scheduler/test_scheduler_entities.py [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+        assert interval_schedule.schedule_id is None
+        assert interval_schedule.name  is None
+        assert interval_schedule.exclusive  is True
+        assert interval_schedule.enabled is False
+        assert interval_schedule.repeat is None
+        assert interval_schedule.process_name is None
+        assert interval_schedule.schedule_type == 3
+        with pytest.raises(AttributeError) as excinfo:
+            assert interval_schedule.day is None
+            assert interval_schedule.time is None
```
Comment (Singhal-Vaibhav) [inline tests/unit/python/foglamp/services/core/scheduler/test_scheduler_entities.py]: Same as above

File: tests/unit/python/foglamp/services/core/scheduler/test_scheduler_entities.py [hunk 7]
```diff
@@ -0,0 +1,104 @@
...
+        assert manual_schedule.schedule_id is None
+        assert manual_schedule.name  is None
+        assert manual_schedule.exclusive  is True
+        assert manual_schedule.enabled is False
+        assert manual_schedule.repeat is None
+        assert manual_schedule.process_name is None
+        assert manual_schedule.schedule_type == 4
+        with pytest.raises(AttributeError) as excinfo:
+            assert manual_schedule.day is None
+            assert manual_schedule.time is None
```
Comment (Singhal-Vaibhav) [inline tests/unit/python/foglamp/services/core/scheduler/test_scheduler_entities.py]: Same as above

**Comment by amarendra-dianomic** [comment]
@Singhal-Vaibhav feedback changed done. Pl check.

**Pull request merged**